ui/src/containers/streams/EditStream.js

```diff
@@ -198,22 +202,97 @@ class EditStream extends Component {
     this.setState({ stream: stream });
   }
 
-  render() {
-    if (this.state.streamUpdated) {
-      return <Redirect to={"/streams/" + this.getStreamId()} />;
+  handleEditorChange(value) {
+    this.setState({
+      editorStream: value,
+      payloadEditorChanges: true,
+    });
+  }
+
+  submitEditorContent() {
+    let parsedStream = undefined;
+    try {
+      parsedStream = JSON.parse(this.state.editorStream, null, 2);
+    } catch (syntaxError) {
+      this.setState({
+        streamUpdated: false,
+        errorMessage: syntaxError.message,
+      });
     }
 
-    const stream = this.state.stream;
+    if (parsedStream !== undefined) {
+      this.props.updateStream(parsedStream).then(() => {
```

Comment:
This function [expects two parameter](https://github.com/DataCater/datacater/blob/main/ui/src/actions/streams.js#L108). This way it is not working.